docs/DeveloperGuide.md

```diff
@@ -280,17 +377,169 @@ Use case ends.
 
 3a. The given index is invalid
 
-> 3a1. AddressBook shows an error message <br>
+>3a1. CMDo shows help message <br>
+> Use case resumes at step 2
+
+  
+### Use case: Find a task
+
+**MSS**
+
+1. User requests to find a task
+2. CMDo shows a list of tasks <br>
+>Use case ends.
+
+**Extensions**
+
+1a. The list is empty
+
+> Use case ends
+
+### Use case: List all tasks
+
+**MSS**
+
+1. User requests to list all tasks
+2. CMDo shows a list of tasks <br>
+>Use case ends.
+
+**Extensions**
+
+1a. The list is empty
+
+### Use case: List done tasks
+
+> Use case ends
+
+**MSS**
+
+1. User requests to list done tasks
+2. CMDo shows a list of done tasks <br>
+>Use case ends.
+
+**Extensions**
+
+1a. The list is empty
+
+> Use case ends
+
+**MSS**
+
+1. User requests to list blocked time slots
+2. CMDo shows a list of blocked time slots <br>
+>Use case ends.
+
+**Extensions**
+
+1a. The list is empty
+
+> Use case ends
+
+### Use case: Mark a task as done
+
+**MSS**
+
+1. User requests to find a task or list all tasks
+2. CMDo shows a list of tasks 
+3. Mark the task done by index 
+4. The task is marked as done and moved to storage<br>
+>Use case ends.
+
+**Extensions**
+
+1a. The list is empty
+
+> Use case ends
+
+3a. The given index is invalid
+
+> 3a1. CMDo shows help message <br>
   Use case resumes at step 2
 
-{More to be added}
+4a. The task will not show up on list or find.
+
+> Use case ends
+
+### Use case: Undo
+
+**MSS**
+
+1. User requests to undo previous action
+2. CMDo requests the user to confirm the action
+3. CMDo undos the previous action <br>
+>Use case ends.
+
+**Extensions**
+
+1a. There are no more previous actions
+
+> Use case ends
+
+2a. User does not confirm the action
+
+> Use case ends
+
+### Use case: Redo
+
+**MSS**
+
+1. User requests to redo previous action
+2. CMDo requests the user to confirm the action
+3. CMDo undos the previous action <br>
```

Comment:
Missing "user confirms action .... " but according to ur system ~ it doesn't exactly request users to confirm am i right?